File: java/src/main/java/be/playing/with/projections/model/projections/CountEventsProjection.java
package be.playing.with.projections.model.projections;

import java.util.List;

import be.playing.with.projections.model.Event;
import be.playing.with.projections.model.Projection;

public class CountEventsProjection implements Projection<Integer> {

  @Override
  public Integer project(List<Event> events) {
    return events.size();
  }

  @Override
  public String buildResultMessage(Integer projectResult) {
    return projectResult + " events counted in the stream";
  }
}

File: kotlin/src/main/kotlin/be/playing/with/projections/model/projections/CountEventsProjection.kt
package be.playing.with.projections.model.projections

import be.playing.with.projections.model.Event
import be.playing.with.projections.model.Projection

class CountEventsProjection : Projection<Int> {

    override fun buildResultMessage(projectResult: Int): String = "$projectResult events counted in the stream"

    override fun project(events: List<Event>): Int = events.size

}

